frontend/src/components/WorkoutTemplateModal/WorkoutTemplateModal.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState, useEffect } from "react";
+import styles from "./WorkoutTemplateModal.module.css";
+
+interface WorkoutTemplateModalProps {
+  closeModal: () => void;
+}
+
+const WorkoutTemplateModal: React.FC<WorkoutTemplateModalProps> = ({
+  closeModal,
+}) => {
+  const [workoutName, setWorkoutName] = useState("");
+  const [searchTerm, setSearchTerm] = useState("");
+  const [exercises, setExercises] = useState<string[]>([]);
+  const [selectedExercises, setSelectedExercises] = useState<string[]>([]);
+  const [filteredExercises, setFilteredExercises] = useState<string[]>([]);
+
+  useEffect(() => {
+    const fetchExercises = async () => {
+      const options = {
+        method: "GET",
+        headers: {
+          "x-rapidapi-key": import.meta.env.VITE_RAPIDAPI_KEY,
+          "x-rapidapi-host": "exercise-db-fitness-workout-gym.p.rapidapi.com",
+        },
+      };
+      const response = await fetch(
+        "https://exercise-db-fitness-workout-gym.p.rapidapi.com/exercises",
+        options
+      );
+      const data = await response.json();
+      setExercises(data.excercises_ids);
+    };
+
+    fetchExercises();
+  }, []);
+
+  useEffect(() => {
+    const normalizedSearchTerm = searchTerm.toLowerCase();
+    setFilteredExercises(
+      exercises.filter((exercise) =>
+        exercise.toLowerCase().includes(normalizedSearchTerm)
+      )
+    );
+  }, [searchTerm, exercises]);
+
+  const handleAddExercise = (exercise: string) => {
+    if (!selectedExercises.includes(exercise)) {
+      setSelectedExercises((prev) => [...prev, exercise]);
+    }
+    setSearchTerm("");
+  };
+
+  const handleRemoveExercise = (exercise: string) => {
+    setSelectedExercises((prev) => prev.filter((ex) => ex !== exercise));
+  };
+
+  const handleCreateTemplate = async () => {
+    if (!workoutName || selectedExercises.length === 0) {
```

Comment:
The default value for workoutName is empty string, not quite familiar with typescript but will !workoutName also check for empty string or just null?

Reply:
!workoutName checks for both empty string and null values